src/js/libgeo/GeoOps.js

```diff
@@ -616,15 +644,170 @@ geoOps.PointOnSegment.updatePosition = function(el) {
     var param = getStateComplexNumber();
     putStateComplexNumber(param); // copy parameter
     var seg = csgeo.csnames[el.args[0]];
-    var start = seg.startpos;
-    var end = seg.endpos;
-    var far = List.sub(end, start);
-    var homog = List.add(start, List.scalmult(param, far));
-    homog = List.normalizeMax(homog);
+
+    var homog = geoOps._helper.projectPointToSegment(seg, param);
     el.homog = General.withUsage(homog, "Point");
 };
 geoOps.PointOnSegment.stateSize = 2;
 
+geoOps._helper.PointOnArcCr = function(arc, P) {
+    var A = arc.startPoint,
+        B = arc.viaPoint,
+        C = arc.endPoint;
+    var cr = List.crossratio3harm(P, A, B, C, List.ii);
+
+    var m = cr.value[0];
+    var n = cr.value[1];
+
+    if (!CSNumber._helper.isAlmostZero(m)) {
+        n = CSNumber.div(n, m);
+        m = CSNumber.real(1);
+    } else {
+        m = CSNumber.div(m, n);
+        n = CSNumber.real(1);
+    }
+
+
+    var prod = m.value.real * n.value.real;
+
+    if (prod >= 0 && prod <= 1) {
```

Comment:
Why this case distinction here based on the *product*? I think this should be 0 ≤ *m*/*n* ≤ 1, which you could write like this if you only care about the real values. If you want to avoid division you have to multiply everything with *n*² like this: 0 ≤ *m*∙*n* ≤ *n*². Could this be the reason why `normalizeMax` doesn't do what you expect?